### ISSUE TITLE
Uncommon TypeError in Safari

### ISSUE THREAD

**Issue by litonico** [catdad/canvas-confetti#109]
First, thanks for this extremely good package!

We run canvas-confetti in production, and have twice seen this error — once on iOS Safari 13.3.1 running on iOS 13.3 (this occurred only once), and once on desktop Safari 13.0.3 running on Mac OSX 10.5 (a single user saw this error four times over the course of five minutes):

```
TypeError t.transferControlToOffscreen is not a function. (In 't.transferControlToOffscreen()', 't.transferControlToOffscreen' is undefined) 
    App/node_modules/canvas-confetti/dist/confetti.module.mjs:103:12 
    App/node_modules/canvas-confetti/dist/confetti.module.mjs:531:22
    [traceback continues through our app code]
```

We're calling the default export directly, not supplying our own canvas:

```
import confetti from "canvas-confetti"
confetti({
  particleCount: 80,
  spread: 70,
  origin: {y: 0.6},
  ticks: 400,
  colors: [
    "#2c2d81",
    "#4547e0",
    "#7e8bff",
    "#d9ddff",
    "#f72750",
    "#b15e4e",
    "#7A7452",
    "#077d55",
    "#1B837c",
    "#cc1d49",
    "#ed9381",
    "#e8da7f",
    "#1dc177",
    "#37a8ae",
  ],
})
```

I haven't been able to reproduce the error. It seems quite rare: it has happened to only two people in the last 3 months, out of approximately 400 people that were both using Safari and saw confetti in the same period.

As far as I know, `window.OffscreenCanvas` should always be `undefined` in Safari, so your conditional check in `canUseWorker` seems like it should be sufficient. Maybe Safari will misreport that on occasion?

**Comment by catdad** [comment]
Oh, that's really weird. As far as I know, Safari isn't even working on Offscreen Canvas. Let me Google around and see what I can find. I can always add an extra check in there, but I am super curious what could be happening.

**Comment by catdad** [comment]
Ugh, so yeah, they are running experiments with an incomplete implementation. Supposedly it's working in Technical Preview, so I'll probably need to dig up a Mac from somewhere. That's frustrating. I can look into adding some more checks, but obviously this is super frustrating since I can't very well test this.

In the meantime, if you want to work around this, you can disable workers in the main canvas like this:

```javascript
const confettiWithoutWorker = confetti.create(null, { useWorker: false, resize: true });
```

**Comment by catdad** [comment]
Safari indeed has an "ImageBitmap and OffscreenCanvas" experiment listed under the Experimental Features in the Develop menu. Once turned on, it enables this incomplete OffscreenCanvas implementation. I added extra checks for `transferControlToOffscreen`, which I think is the only thing not implemented yet that I do actually try to use. If you run into any more issues like this, please let me know.

This fix has been published in v1.1.2

**Comment by litonico** [comment]
Thank you so much!
